app/jobs/user_session_jobs/schedule_daily_messages_job.rb

```diff
@@ -0,0 +1,130 @@
+    user = user_session.user
+    session = user_session.session
+
+    # Find all accessible question groups
+    accessible_question_groups = calculate_accessible_question_groups_for_user(session.question_groups.order(:position), user_session)
+    today_scheduled_question_groups = select_questions_groups_scheduled_for_today(accessible_question_groups)
+
+    questions_to_be_send_today = []
+
+    # Find all questions scheduled for today in groups
+    today_scheduled_question_groups.each do |question_group|
+      last_answer = last_answer_in_question_group(question_group, user_session)
+      questions_per_day = question_group.sms_schedule['questions_per_day'] || 1
+      last_question_index = question_group.questions.order(position: :desc).first.position
+      first_question_index = question_group.questions.order(position: :asc).first.position
+
+      question_positions = if !last_answer || last_answer.question.position == last_question_index
+                             base_range = (first_question_index..last_question_index).to_a
+                             positions_range = base_range
+                             (questions_per_day / base_range.length.to_f).ceil.to_i.times do
+                               positions_range += base_range
+                             end
+                             positions_range
+                           elsif last_answer.question.position + questions_per_day > last_question_index
+                             base_range = (first_question_index..last_question_index).to_a
+                             from_current_question_range = ((last_answer.question.position + 1)..last_question_index).to_a
+                             ((questions_per_day / base_range.length.to_f).ceil - 1).to_i.times do
+                               from_current_question_range += base_range
+                             end
+                             from_current_question_range
+                           else
+                             ((last_answer.question.position + 1)..last_question_index).to_a
+                           end
+
+      positions_to_be_send = question_positions.first(questions_per_day)
+
+      positions_to_be_send.each_with_index do |position, index|
+        question = question_group.questions.find_by(position: position)
+        time_to_send = calculate_question_sending_time(get_proper_sending_period(user_session.user_intervention, question_group.sms_schedule),
+                                                       questions_per_day, index)
+        questions_to_be_send_today << { question: question, time_to_send: time_to_send }
+      end
+    end
+
+    # Remove all Information questions if there is any question requiring attention
+    any_answer_expected = questions_to_be_send_today.map { |elem| elem[:question].type }.include?('Question::Sms')
+    questions_to_be_send_today.reject! { |elem| elem[:question].type.match('Question::SmsInformation') } if any_answer_expected
```

Comment:
I don't see this requirement in the task, could you update the Jira's task to be corresponding to this?

Reply:
This is part, that was discussed directly with client. I will update Jira task as you suggested 👍